src/api-service/__app__/agent_events/__init__.py

```diff
@@ -41,13 +41,37 @@ def get_node_checked(machine_id: UUID) -> Node:
     return node
 
 
-def on_state_update(machine_id: UUID, state: NodeState) -> func.HttpResponse:
+def on_state_update(
+    machine_id: UUID,
+    state_update: NodeStateUpdate,
+) -> func.HttpResponse:
+    state = state_update.state
     node = get_node_checked(machine_id)
 
     if state == NodeState.init or node.state not in NodeState.ready_for_reset():
         if node.state != state:
             node.state = state
             node.save()
+
+            if state == NodeState.setting_up:
+                # This field will be required in the future.
+                # For now, it is optional for back compat.
+                if state_update.data:
+                    for task_id in state_update.data.tasks:
+                        task = get_task_checked(task_id)
+                        task.state = TaskState.setting_up
```

Comment:
won't this conflict with line 95 when we set the state to running in the case where we have multiple nodes running the same task

Reply:
Yes. I guess the worst problem is that we could go from `setting_up`, to `running`, and back to `setting_up`, if we have e.g. a task with `vm_count == 2`. Also, I think that problem is not unique to this new state.

Regardless, maybe we should just remove the `Task` state, but leave `setting_up` as a `NodeTaskState`.

Reply:
my thoughts exactly

Reply:
Handled for now by only updating if we aren't `running`.